subprojects/core/src/main/java/org/gradle/internal/service/scopes/ProjectScopeServices.java

```diff
@@ -115,23 +119,29 @@
 public class ProjectScopeServices implements ServiceRegistrationProvider {
 
     public static CloseableServiceRegistry create(
-        ServiceRegistry parent,
+        ServiceRegistry buildServices,
         ProjectInternal project,
         Factory<LoggingManagerInternal> loggingManagerInternalFactory
     ) {
+        BuildLayout buildLayout = buildServices.get(BuildLayout.class);
         return ServiceRegistryBuilder.builder()
             .scope(Scope.Project.class)
             .displayName("project services")
-            .parent(parent)
-            .provider(new ProjectScopeServices(project, loggingManagerInternalFactory))
+            .parent(buildServices)
+            .provider(new ProjectScopeServices(buildLayout.getSettingsDir(), project.getProjectDir(), project, loggingManagerInternalFactory))
             .provider(new WorkerSharedProjectScopeServices(project.getProjectDir()))
             .build();
     }
 
+    private final File settingsDir;
+    private final File projectDir;
```

Comment:
❌ These two fields are unnecessary and can be removed given the `project` is already there.